vignettes/fims-path-maturity.Rmd

```diff
@@ -0,0 +1,549 @@
+
+
+Linking to the `fims::Information` class allows us to register maturity parameters and specify whether or not they are random effects:
+
+```{Rcpp, eval = FALSE}
+//d0 is a shared pointer that points to fims::Information
+std::shared_ptr<fims::Information<TMB_FIMS_REAL_TYPE> > d0 =
+        fims::Information<TMB_FIMS_REAL_TYPE>::GetInstance();
+ if (this->median.estimated) {
+      if (this->median.is_random_effect) {
+        /*
+         if median is estimated and a random effect, 
+         the median value from LogisticMaturityInterface (lm0->median) 
+         is passed to the Information member function RegisterRandomEffect
+        */
+        d0->RegisterRandomEffect(lm0->median);
+      } else {
+        /*
+         if median is estimated and not a random effect, 
+         the median value from LogisticMaturityInterface (lm0->median)
+         is passed to the Information member function RegisterParameter
+        d0->RegisterParameter(lm0->median);
+      }
+    }
+```
+
+
+Linking to the `fims::LogisticMaturity` class allows us to link the values input from R with the values used in model calculation:
+```{Rcpp, eval = FALSE}
+//lm0 is a shared pointer that points to fims::LogisticMaturity
+std::shared_ptr<fims::LogisticMaturity<TMB_FIMS_REAL_TYPE> > lm0 =
+        std::make_shared<fims::LogisticMaturity<TMB_FIMS_REAL_TYPE> >();
+ 
+/*
+ the median value from LogisticMaturity (lm0->median)
+ equals the median value from LogisticMaturityInterface (this->median.value)
+ */
+lm0->median = this->median.value;
+```
+
+We can also link these two pointers together so that the **fims::Information** class links up with the **fims::LogisticMaturity**, but more details on this later.
+```{Rcpp, eval = FALSE}
+/*
+ the maturity_models pointer from Information that matches the id of the fims::LogisticMaturity class is equal to the pointer to fims::LogisticMaturity 
+ */
+d0->maturity_models[lm0->id] = lm0;
+```
+
+
+The `add_to_fims_tmb()` function repeats code four times to track the estimated value of parameters along with their first, second, and third derivatives. 
+
+
+```{r image-path-maturity-2, echo = FALSE, message=FALSE, out.width = '100%'}
+knitr::include_graphics("figures/fims-path-maturity-2.png")
```

Comment:
Thank you for making all these graphics! The visuals are so helpful.

Reply:
+1 This graphic deserves a 🏆 and helped me wrap my brain around a lot of missing pieces. The color coding of the parts and the ⬇️s were really helpful. I wonder though if the color of all the arrows should be orange?

Reply:
I increased the line size, thanks for the suggestion!